quickcheck-dynamic/src/Test/QuickCheck/StateModel.hs

```diff
@@ -239,23 +247,6 @@ class (forall a. Show (Action state a), Monad m) => RunModel state m where
   monitoringFailure :: state -> Action state a -> LookUp m -> Error state -> Property -> Property
   monitoringFailure _ _ _ _ prop = prop
 
-computePostcondition
-  :: forall m state a
-   . RunModel state m
-  => (state, state)
-  -> ActionWithPolarity state a
-  -> LookUp m
-  -> Either (Error state) (Realized m a)
-  -> PostconditionM m Bool
-computePostcondition ss (ActionWithPolarity a p) l r
-  | p == PosPolarity = case r of
-      Right ra -> postcondition ss a l ra
-      -- NOTE: this is actually redundant as this handled
-      -- at the call site for this function, but this is
-      -- good hygiene?
-      Left _ -> pure False
-  | otherwise = postconditionOnFailure ss a l r
```

Comment:
I have a slight gripe with removing `computePostcondition` as a thing to be able to grep for. We have these `computeX` functions because the underlying runtime has this extra semantics around negative actions that isn't always obvious, so it's good to have one place where it says "here is the postcondition we actually run"..

Now, that does not mean that the current version of the code is super clear. I just wish we can retain this _particular_ property of being able to easily grep for the semantics around e.g. negative vs positive actions.

Reply:
I also liked the fact the `evaluatePostcondition` function is very abstract and only ties the 2 monads together, `PostconditionM` and `PropertyM`

Reply:
Also, by `grep`ping you mean grepping for `NegPolarity`?

Reply:
> I also liked the fact the `evaluatePostcondition` function is very abstract and only ties the 2 monads together, `PostconditionM` and `PropertyM`

That I also like!

> Also, by grepping you mean grepping for NegPolarity?

No, I mean for like `computeNextState`, `computePrecondition`, `computePostcondition` as a way of reminding yourself of the semantics. But I think that's quite a weak argument from my point.

Reply:
🤔

Reply:
Ugh... I think I managed to make myself more confused than anything with this comment. Let's forget I said anything about this! I think this change is fine!